test/tap/groups/lint_groups_json.py

```
@@ -0,0 +1,119 @@
+        errors.append("Top-level value must be a JSON object")
+        for e in errors:
+            print(f"ERROR: {e}", file=sys.stderr)
+        return 1
+
+    # Check 2: first and last lines
+    if not lines or lines[0].strip() != "{":
+        errors.append("Line 1: must be '{'")
+    if lines[-1].strip() == "":
+        # Allow trailing newline — check second-to-last
+        if len(lines) < 2 or lines[-2].strip() != "}":
+            errors.append(f"Last non-empty line: must be '}}'")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)